### PR TITLE
index: map each release with compatible nrf-sdk version

### PR DESCRIPTION
**Pull request by FilipZajdel**
VSC-2567

Changes:
- From now on each add-on release is accompanied with the nRF Connect SDK revision it is compatible with. This removes the `compatibleNcs` array previously utilized for representing similar relationship.
- Each release needs to be manually announced in the Add-ons index by modifying particular customer's index file.

### CODE REVIEW

File: site/src/schema.ts [hunk 1]
```diff
@@ -93,16 +93,24 @@ export const appMetadataSchema = {
                 "By default, the VS Code extension will assume that there's just a single application sitting at the root of the repo.",
             ].join('\n\n'),
         },
-        compatibleNcs: {
+        releases: {
             type: 'array',
             items: {
-                type: 'string'
+                type: 'object',
+                properties: {
```
Comment (bencefr) [inline site/src/schema.ts:100]: Would be nice to extend the schema with descriptions, after all externals are supposed to contribute to the JSON content.

File: site/src/app/ReleasesDropDownList.tsx [hunk 2]
```diff
@@ -12,6 +12,7 @@ interface Props {
 };
 
 function ReleasesDropDownList({ app, onReleaseChosen }: Props): JSX.Element {
+  console.log(app.releases ?? `${app.name} doesn't have releases`)
```
Comment (bencefr) [inline site/src/app/ReleasesDropDownList.tsx:15]: Perhaps remove console.log.

File: index/circuitdojo.json [hunk 3]
```diff
@@ -9,7 +9,14 @@
             "kind": "sample",
             "tags": ["lte"],
             "apps": "samples/*",
-            "compatibleNcs": ["v2.4.2"]
+            "releases": [
+                {
+                    "date": "2023-12-20T20:37:28Z",
+                    "name": "2.4.x",
+                    "tag": "v2.4.x",
+                    "sdk": "v2.4.1"
```
Comment (bencefr) [inline index/circuitdojo.json:17]: All over it looks very inconsistent that the SDK version sometimes starts with `v` sometimes doesn't. How is it expected? Even if it works both ways, this should probably be cleaned up.

**Review by bencefr — APPROVED** [review]
Apart from my minor comments the change looks alright, although there's a conflicting request already from Golioth. Please comment or suggest on their PR to smoothen the ride.

File: index/circuitdojo.json (hunk 3 above)
Comment (greg-fer) [reply index/circuitdojo.json:17]: `v` should be used for consistency with the docs throughout all of the releases, if you ask me.

**Review by greg-fer — COMMENTED** [review]
Please add `v` to SDK releases.

File: index/circuitdojo.json (hunk 3 above)
Comment (bencefr) [reply index/circuitdojo.json:17]: This is only data here, whether the version contains `v` or not here is not relevant if it's takes care of when we display it, but consistency and data integrity is important.

File: resources/output_schema.json [hunk 4]
```diff
@@ -153,12 +153,16 @@
...
+                                "sdk": {
+                                    "type": "string"
                                 }
                             },
                             "required": [
                                 "tag",
                                 "name",
-                                "date"
+                                "date",
+                                "sdk"
```
Comment (bencefr) [inline resources/output_schema.json:165]: Hold on, is `sdk` required? In https://github.com/nrfconnect/vscode-nrf-connect/pull/4020 it is displayed optionally, and that perhaps makes sense.

Comment (FilipZajdel) [reply resources/output_schema.json:165]: I was not able to figure out the sdk versions of several releases. I'll do it once again though.

**Comment by FilipZajdel** [comment]
@greg-fer can you have a look at the descriptions in schema I just have added, please?

**Review by bencefr — APPROVED** [review]

**Review by greg-fer — APPROVED** [review]

**Pull request merged**